### PR TITLE
Refactor volcano iii

### PR DESCRIPTION
**Pull request by mschwoer**
move the gene to protein id mapping to DataSet, thus simplifying LLM part

### CODE REVIEW

File: alphastats/DataSet.py [hunk 1]
```diff
@@ -94,6 +94,21 @@ def __init__(
...
 
+        self._gene_name_to_protein_id_map = (
+            {
+                k: v
+                for k, v in dict(
+                    zip(
+                        self.rawinput[self._gene_names].tolist(),
+                        self.rawinput[self.index_column].tolist(),
+                    )
+                ).items()
```
Comment (JuliaS92) [inline alphastats/DataSet.py:105]: This is not necessarily unique, and should ideally raise an error in some of our test-data sets that contain isoform ids. E.g. TPM1 occurs 5 times in testfiles/maxquant/proteinGroups.txt with different base Protein IDs.

File: tests/test_DataSet.py [hunk 2]
```diff
@@ -776,6 +776,18 @@ def test_multicova_analysis_invalid_covariates(self):
...
+            self.obj._get_protein_id_for_gene_name("MADE_UP_GENE"), "MADE_UP_GENE"
+        )
+        self.assertEqual(
+            self.obj._get_protein_id_for_gene_name("ALDOC"),
+            "P09972;A0A024QZ64;A8MVZ9;B7Z3K9;B7Z1N6;B7Z3K7;J3KSV6;J3QKP5;C9J8F3;B7Z1Z9;J3QKK1;B7Z1H6;K7EKH5;B7Z1L5",
+        )
+        self.assertEqual(
+            self.obj._get_protein_id_for_gene_name("FCGRT"), "P55899;M0R0A9;A0A024QZI2"
+        )
+
```
Comment (JuliaS92) [inline tests/test_DataSet.py:790]: TODO: This needs to cover cases akin to what has been deleted, and needs to be tested on an artificial frame, as different searche engine outputs cover different cases.

**Review by JuliaS92 — APPROVED** [review]
LGTM
Anything regarding ids needs to be addressed in a separate PR.

File: tests/test_DataSet.py (hunk 2 above)
Comment (mschwoer) [reply tests/test_DataSet.py:790]: not sure what you mean .. all the branches of the method are tested .. a artifical frame would make the test more easy to read, but that's out current testing strategy ;-)

**Pull request merged**